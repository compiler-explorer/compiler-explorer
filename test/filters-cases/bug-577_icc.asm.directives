.LNDBG_TX:
# mark_description "Intel(R) C Intel(R) 64 Compiler for applications running on Intel(R) 64, Version 18.0.0.128 Build 20170811";
# mark_description "-g -o /tmp/compiler-explorer-compiler1171120-54-1t2ppc6.m4k6/output.s -masm=intel -S -gxx-name=/opt/compiler";
# mark_description "-explorer/gcc-6.3.0/bin/g++";
..TXTST0:
.L_2__routine_start__Z6squarei_0:
# -- Begin  _Z6squarei
# mark_begin;

# --- square(int)
_Z6squarei:
# parameter 1(num): edi
..B1.1:                         # Preds ..B1.0
                                # Execution count [0.00e+00]
..___tag_value__Z6squarei.2:
..L3:
                                                          #2.21
..LN0:
        push      rbp                                           #2.21
..LN1:
        mov       rbp, rsp                                      #2.21
..LN2:
        sub       rsp, 16                                       #2.21
..LN3:
        mov       DWORD PTR [-16+rbp], edi                      #2.21
..LN4:
                                # LOE rbx rbp rsp r12 r13 r14 r15 rip
..B1.5:                         # Preds ..B1.1
                                # Execution count [0.00e+00]
# Begin ASM
..LN5:
# Begin ASM
label:
# End ASM                                                       #3.0
..LN6:
# End ASM
                                # LOE rbx rbp rsp r12 r13 r14 r15 rip
..B1.4:                         # Preds ..B1.5
                                # Execution count [0.00e+00]
..LN7:
        mov       eax, DWORD PTR [-16+rbp]                      #4.18
..LN8:
        imul      eax, DWORD PTR [-16+rbp]                      #4.18
..LN9:
        leave                                                   #4.18
..LN10:
        ret                                                     #4.18
..LN11:
                                # LOE
..LN12:
# mark_end;
..LN_Z6squarei.13:
.LN_Z6squarei:
# -- End  _Z6squarei
// -- Begin DWARF2 SEGMENT .debug_info
.debug_info_seg:
//      DW_TAG_compile_unit:
//      DW_AT_comp_dir:
//      DW_AT_name:
//      DW_AT_producer:
//      DW_AT_language:
//      DW_AT_use_UTF8:
//      DW_AT_low_pc:
//      DW_AT_high_pc:
//      DW_AT_stmt_list:
//      DW_TAG_namespace:
//      DW_AT_name:
//      DW_TAG_namespace:
//      DW_AT_name:
//      DW_TAG_namespace:
//      DW_AT_name:
//      DW_TAG_base_type:
//      DW_AT_byte_size:
//      DW_AT_encoding:
//      DW_AT_name:
//      DW_TAG_subprogram:
//      DW_AT_decl_line:
//      DW_AT_decl_file:
//      DW_AT_type:
//      DW_AT_name:
//      DW_AT_low_pc:
//      DW_AT_high_pc:
//      DW_AT_external:
//      DW_TAG_formal_parameter:
//      DW_AT_decl_line:
//      DW_AT_decl_file:
//      DW_AT_type:
//      DW_AT_name:
//      DW_AT_location:
// -- Begin DWARF2 SEGMENT .debug_line
.debug_line_seg:
// -- Begin DWARF2 SEGMENT .debug_abbrev
.debug_abbrev_seg:
// -- Begin DWARF2 SEGMENT .debug_frame
.debug_frame_seg:
// -- Begin DWARF2 SEGMENT .debug_str
.debug_str_seg:
// -- Begin DWARF2 SEGMENT .eh_frame
.eh_frame_seg:
.LNDBG_TXe:
# End
