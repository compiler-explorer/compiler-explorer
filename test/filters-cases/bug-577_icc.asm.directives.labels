# mark_description "Intel(R) C Intel(R) 64 Compiler for applications running on Intel(R) 64, Version 18.0.0.128 Build 20170811";
# mark_description "-g -o /tmp/compiler-explorer-compiler1171120-54-1t2ppc6.m4k6/output.s -masm=intel -S -gxx-name=/opt/compiler";
# mark_description "-explorer/gcc-6.3.0/bin/g++";
# -- Begin  _Z6squarei
# mark_begin;

# --- square(int)
_Z6squarei:
# parameter 1(num): edi
                                # Execution count [0.00e+00]
                                                          #2.21
        push      rbp                                           #2.21
        mov       rbp, rsp                                      #2.21
        sub       rsp, 16                                       #2.21
        mov       DWORD PTR [-16+rbp], edi                      #2.21
                                # LOE rbx rbp rsp r12 r13 r14 r15 rip
                                # Execution count [0.00e+00]
# Begin ASM
# Begin ASM
# End ASM                                                       #3.0
# End ASM
                                # LOE rbx rbp rsp r12 r13 r14 r15 rip
                                # Execution count [0.00e+00]
        mov       eax, DWORD PTR [-16+rbp]                      #4.18
        imul      eax, DWORD PTR [-16+rbp]                      #4.18
        leave                                                   #4.18
        ret                                                     #4.18
                                # LOE
# mark_end;
# -- End  _Z6squarei
// -- Begin DWARF2 SEGMENT .debug_info
//      DW_TAG_compile_unit:
//      DW_AT_comp_dir:
//      DW_AT_name:
//      DW_AT_producer:
//      DW_AT_language:
//      DW_AT_use_UTF8:
//      DW_AT_low_pc:
//      DW_AT_high_pc:
//      DW_AT_stmt_list:
//      DW_TAG_namespace:
//      DW_AT_name:
//      DW_TAG_namespace:
//      DW_AT_name:
//      DW_TAG_namespace:
//      DW_AT_name:
//      DW_TAG_base_type:
//      DW_AT_byte_size:
//      DW_AT_encoding:
//      DW_AT_name:
//      DW_TAG_subprogram:
//      DW_AT_decl_line:
//      DW_AT_decl_file:
//      DW_AT_type:
//      DW_AT_name:
//      DW_AT_low_pc:
//      DW_AT_high_pc:
//      DW_AT_external:
//      DW_TAG_formal_parameter:
//      DW_AT_decl_line:
//      DW_AT_decl_file:
//      DW_AT_type:
//      DW_AT_name:
//      DW_AT_location:
// -- Begin DWARF2 SEGMENT .debug_line
// -- Begin DWARF2 SEGMENT .debug_abbrev
// -- Begin DWARF2 SEGMENT .debug_frame
// -- Begin DWARF2 SEGMENT .debug_str
// -- Begin DWARF2 SEGMENT .eh_frame
# End