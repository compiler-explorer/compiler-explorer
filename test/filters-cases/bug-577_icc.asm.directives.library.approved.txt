{
  "asm": [
    {
      "labels": [],
      "source": null,
      "text": ".LNDBG_TX:"
    },
    {
      "labels": [],
      "source": null,
      "text": "# mark_description \"Intel(R) C Intel(R) 64 Compiler for applications running on Intel(R) 64, Version 18.0.0.128 Build 20170811\";"
    },
    {
      "labels": [],
      "source": null,
      "text": "# mark_description \"-g -o /tmp/compiler-explorer-compiler1171120-54-1t2ppc6.m4k6/output.s -masm=intel -S -gxx-name=/opt/compiler\";"
    },
    {
      "labels": [],
      "source": null,
      "text": "# mark_description \"-explorer/gcc-6.3.0/bin/g++\";"
    },
    {
      "labels": [],
      "source": null,
      "text": "..TXTST0:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".L_2__routine_start__Z6squarei_0:"
    },
    {
      "labels": [],
      "source": null,
      "text": "# -- Begin  _Z6squarei"
    },
    {
      "labels": [],
      "source": null,
      "text": "# mark_begin;"
    },
    {
      "labels": [],
      "source": null,
      "text": ""
    },
    {
      "labels": [],
      "source": null,
      "text": "# --- square(int)"
    },
    {
      "labels": [],
      "source": null,
      "text": "_Z6squarei:"
    },
    {
      "labels": [],
      "source": null,
      "text": "# parameter 1(num): edi"
    },
    {
      "labels": [],
      "source": null,
      "text": "..B1.1:                         # Preds ..B1.0"
    },
    {
      "labels": [],
      "source": null,
      "text": "                                # Execution count [0.00e+00]"
    },
    {
      "labels": [],
      "source": null,
      "text": "..___tag_value__Z6squarei.2:"
    },
    {
      "labels": [],
      "source": null,
      "text": "..L3:"
    },
    {
      "labels": [],
      "source": null,
      "text": "                                                          #2.21"
    },
    {
      "labels": [],
      "source": null,
      "text": "..LN0:"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 2
      },
      "text": "        push      rbp                                           #2.21"
    },
    {
      "labels": [],
      "source": null,
      "text": "..LN1:"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 2
      },
      "text": "        mov       rbp, rsp                                      #2.21"
    },
    {
      "labels": [],
      "source": null,
      "text": "..LN2:"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 2
      },
      "text": "        sub       rsp, 16                                       #2.21"
    },
    {
      "labels": [],
      "source": null,
      "text": "..LN3:"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 2
      },
      "text": "        mov       DWORD PTR [-16+rbp], edi                      #2.21"
    },
    {
      "labels": [],
      "source": null,
      "text": "..LN4:"
    },
    {
      "labels": [],
      "source": null,
      "text": "                                # LOE rbx rbp rsp r12 r13 r14 r15 rip"
    },
    {
      "labels": [],
      "source": null,
      "text": "..B1.5:                         # Preds ..B1.1"
    },
    {
      "labels": [],
      "source": null,
      "text": "                                # Execution count [0.00e+00]"
    },
    {
      "labels": [],
      "source": null,
      "text": "# Begin ASM"
    },
    {
      "labels": [],
      "source": null,
      "text": "..LN5:"
    },
    {
      "labels": [],
      "source": null,
      "text": "# Begin ASM"
    },
    {
      "labels": [],
      "source": null,
      "text": "label:"
    },
    {
      "labels": [],
      "source": null,
      "text": "# End ASM                                                       #3.0"
    },
    {
      "labels": [],
      "source": null,
      "text": "..LN6:"
    },
    {
      "labels": [],
      "source": null,
      "text": "# End ASM"
    },
    {
      "labels": [],
      "source": null,
      "text": "                                # LOE rbx rbp rsp r12 r13 r14 r15 rip"
    },
    {
      "labels": [],
      "source": null,
      "text": "..B1.4:                         # Preds ..B1.5"
    },
    {
      "labels": [],
      "source": null,
      "text": "                                # Execution count [0.00e+00]"
    },
    {
      "labels": [],
      "source": null,
      "text": "..LN7:"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 4
      },
      "text": "        mov       eax, DWORD PTR [-16+rbp]                      #4.18"
    },
    {
      "labels": [],
      "source": null,
      "text": "..LN8:"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 4
      },
      "text": "        imul      eax, DWORD PTR [-16+rbp]                      #4.18"
    },
    {
      "labels": [],
      "source": null,
      "text": "..LN9:"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 4
      },
      "text": "        leave                                                   #4.18"
    },
    {
      "labels": [],
      "source": null,
      "text": "..LN10:"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 4
      },
      "text": "        ret                                                     #4.18"
    },
    {
      "labels": [],
      "source": null,
      "text": "..LN11:"
    },
    {
      "labels": [],
      "source": null,
      "text": "                                # LOE"
    },
    {
      "labels": [],
      "source": null,
      "text": "..LN12:"
    },
    {
      "labels": [],
      "source": null,
      "text": "# mark_end;"
    },
    {
      "labels": [],
      "source": null,
      "text": "..LN_Z6squarei.13:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LN_Z6squarei:"
    },
    {
      "labels": [],
      "source": null,
      "text": "# -- End  _Z6squarei"
    },
    {
      "labels": [],
      "source": null,
      "text": "// -- Begin DWARF2 SEGMENT .debug_info"
    },
    {
      "labels": [],
      "source": null,
      "text": ".debug_info_seg:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_TAG_compile_unit:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_comp_dir:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_name:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_producer:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_language:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_use_UTF8:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_low_pc:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_high_pc:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_stmt_list:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_TAG_namespace:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_name:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_TAG_namespace:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_name:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_TAG_namespace:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_name:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_TAG_base_type:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_byte_size:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_encoding:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_name:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_TAG_subprogram:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_decl_line:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_decl_file:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_type:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_name:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_low_pc:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_high_pc:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_external:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_TAG_formal_parameter:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_decl_line:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_decl_file:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_type:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_name:"
    },
    {
      "labels": [],
      "source": null,
      "text": "//      DW_AT_location:"
    },
    {
      "labels": [],
      "source": null,
      "text": "// -- Begin DWARF2 SEGMENT .debug_line"
    },
    {
      "labels": [],
      "source": null,
      "text": ".debug_line_seg:"
    },
    {
      "labels": [],
      "source": null,
      "text": "// -- Begin DWARF2 SEGMENT .debug_abbrev"
    },
    {
      "labels": [],
      "source": null,
      "text": ".debug_abbrev_seg:"
    },
    {
      "labels": [],
      "source": null,
      "text": "// -- Begin DWARF2 SEGMENT .debug_frame"
    },
    {
      "labels": [],
      "source": null,
      "text": ".debug_frame_seg:"
    },
    {
      "labels": [],
      "source": null,
      "text": "// -- Begin DWARF2 SEGMENT .debug_str"
    },
    {
      "labels": [],
      "source": null,
      "text": ".debug_str_seg:"
    },
    {
      "labels": [],
      "source": null,
      "text": "// -- Begin DWARF2 SEGMENT .eh_frame"
    },
    {
      "labels": [],
      "source": null,
      "text": ".eh_frame_seg:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LNDBG_TXe:"
    },
    {
      "labels": [],
      "source": null,
      "text": "# End"
    }
  ],
  "labelDefinitions": {
    "_Z6squarei": 11
  }
}
